docs/usage/validators.md

```diff
@@ -1,375 +1,244 @@
-    int_list: SortedList[int]
-    name_list: SortedList[Name]
-
-
-print(DemoModel(int_list=[3, 2, 1], name_list=['adrian g', 'David']))
-#> int_list=[1, 2, 3] name_list=['Adrian G', 'David']
-```
-
-### Wrap validators
-
-Wrap validators are useful for cases where you want to validate a value, but also want to return a default value if validation fails.
-
-```py
-from datetime import datetime
-
-from typing_extensions import Annotated
-
-from pydantic import BaseModel, ValidationError, WrapValidator
-
-
-def validate_timestamp(v, handler):
-    if v == 'now':
-        # we don't want to bother with further validation, just return the new value
-        return datetime.now()
-    try:
-        return handler(v)
-    except ValidationError:
-        # validation failed, in this case we want to return a default value
-        return datetime(2000, 1, 1)
-
-
-MyTimestamp = Annotated[datetime, WrapValidator(validate_timestamp)]
-
-
-class Model(BaseModel):
-    a: MyTimestamp
 
+try:
+    UserModel(id='abc', name='John Doe')
+except ValidationError as e:
+    print(e)
+    """
+    1 validation error for UserModel
+    id
+      Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='abc', input_type=str]
+    """
 
-print(Model(a='now').a)
-#> 2032-01-02 03:04:05.000006
-print(Model(a='invalid').a)
-#> 2000-01-01 00:00:00
+try:
+    UserModel(id=1, name='John Doe!')
+except ValidationError as e:
+    print(e)
+    """
+    1 validation error for UserModel
+    name
+      Assertion failed, name must be alphanumeric
+    assert False [type=assertion_error, input_value='John Doe!', input_type=str]
+    """
 ```
 
-## Reuse validators
-
-Occasionally, you will want to use the same validator on multiple fields/models (e.g. to
-normalize some input data). The "naive" approach would be to write a separate function,
-then call it from multiple decorators.  Obviously, this entails a lot of repetition and
-boiler plate code. To circumvent this, the `allow_reuse` parameter has been added to
-`pydantic.validator` in **v1.2** (`False` by default):
-
-```py
-from pydantic import BaseModel, field_validator
-
-
-def normalize(name: str) -> str:
-    return ' '.join((word.capitalize()) for word in name.split(' '))
-
-
-class Producer(BaseModel):
-    name: str
-
-    # validators
-    normalize_name = field_validator('name')(normalize)
-
-
-class Consumer(BaseModel):
-    name: str
-
-    # validators
-    normalize_name = field_validator('name')(normalize)
-
+A few things to note on validators:
 
-jane_doe = Producer(name='JaNe DOE')
-john_doe = Consumer(name='joHN dOe')
-assert jane_doe.name == 'Jane Doe'
-assert john_doe.name == 'John Doe'
-```
+* validators are "class methods", so the first argument value they receive is the `UserModel` class, not an instance of `UserModel`. We recommend you use the `@classmethod` decorator on them below the `@field_validator` decorator to get proper type checking.
```

Comment:
I think it's maybe worth drawing the distinction between field_validator and model_validator, or somehow accounting for the fact that `@model_validator(mode='after')` can be used on instance methods. I'm not saying we need to say that here in full detail, but I think maybe we should phrase this point in a way that feels less in-conflict with that fact.

Reply:
(One way to achieve that might be to replace "validators" with "`field_validator`s" in this text